### ISSUE TITLE
JSON assertion for `lessThan` & `moreThan` operators is not working properly.

### ISSUE THREAD

**Issue by chussenot** [DataDog/terraform-provider-datadog#1153]
### Terraform Version

$ terraform -v
Terraform v0.13.4
+ provider registry.terraform.io/datadog/datadog v2.12.1
+ provider registry.terraform.io/hashicorp/vault v2.22.0

### Affected Resource(s)

Please list the resources as a list, for example:

-   datadog_synthetics_test

### Terraform Configuration Files

```csv
# list.csv
name,id,platform,threshold
Wallmart,85124,US,200000
```

```hcl
variable filename {
  default = "list.csv"
}

locals {
  contents = csvdecode(file(var.filename))
}

output "contents" { value = local.contents }

resource "datadog_synthetics_test" "golden" {
  for_each = { for item in local.contents : item.name => item }

  assertion {
    operator = "is"
    target   = "200"
    type     = "statusCode"
  }

  assertion {
    operator = "validatesJSONPath"
    type     = "body"
    targetjsonpath {
      operator    = "moreThan"
      targetvalue = each.value.threshold
      jsonpath    = "content.online"
    }
  }

  device_ids = []
  locations = [
    "pl:private-ecs-prd-df786df78",
  ]

  name = " ${each.value.name} ${each.value.platform} (${each.value.id}) should have more than ${each.value.threshold} products online"
  message = "${each.value.id}"

  options = {
    "min_failure_duration" = "120"
    "min_location_failed"  = "1"
    "tick_every"           = "3600"
  }
  request = {
    "method"  = "GET"
    "url"     = "http://ms.domain.com/api/sellers/1236800/products/stats"
  }
  status  = "live"
  subtype = "http"
  tags = [
    "type:synthetic",
    "managed_by:terraform",
    "env:prd"
  ]
  type = "api"

  request_basicauth {
    password = "who?"
    username = "me" 
  }

}
```

### Debug Output

```bash
$ tf apply
data.terraform_remote_state.remote_state: Refreshing state...
data.vault_generic_secret.datadog: Refreshing state... [id=support/infra/data
datadog_synthetics_test.golden["A Super Mall"]: Refreshing state... [id=t4g-u

Warning: Interpolation-only expressions are deprecated

  on golden-list.tf line 32, in resource "datadog_synthetics_test" "golden":
  32:   message = "${each.value.id}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.


Error: rpc error: code = Unavailable desc = transport is closing
```

### Expected Behavior

Create the right assertion on the probe. By the way the `is` assertion work fine.

```
  assertion {
    operator = "validatesJSONPath"
    type     = "body"
    targetjsonpath {
      operator    = "is"
      targetvalue = each.value.threshold
      jsonpath    = "content.online"
    }
  }
  ``


### Actual Behavior

It's impossible to terrafom probes with this kind of assertions. Is this due to the management of the API v1 in the go code?

**Comment by chussenot** [comment]
@therve https://github.com/DataDog/terraform-provider-datadog/issues?q=+is%3Aissue+author%3A%40me+ ... it's me again sorry but the provider is not working with all the operators.

**Comment by chussenot** [comment]
It's fixed in version `3.2.0` of the provider. the code above has been adapted to do the same thing with blocks. Thx @therve for your support.

```
locals {
  contents = csvdecode(file(var.filename))
}

output "contents" { value = local.contents }

resource "datadog_synthetics_test" "seller_products_volume_check" {
  for_each = { for item in local.contents : item.name => item }

  assertion {
    operator = "is"
    target   = "200"
    type     = "statusCode"
  }

  assertion {
    operator = "validatesJSONPath"
    type     = "body"
    targetjsonpath {
      operator    = "moreThan"
      targetvalue = tonumber(each.value.threshold)
      jsonpath    = "content.online"
    }
  }

  device_ids = []
  locations = [
    "pl:private-ecs-prd-345jk5h345h",
  ]

  name = " ${each.value.name} ${each.value.platform} (${each.value.id}) should have more than ${each.value.threshold} products online"
  message = "${each.value.id}"

  options_list {
    min_failure_duration = "120"
    min_location_failed  = "1"
    tick_every           = "3600"
  }

  request_definition {
    method  = "GET"
    url     = "http://ms.catalog.domain.com/api/sellers/1236800/products/stats"
  }

  status  = "live"
  subtype = "http"
  tags = [
    "type:synthetic",
    "env:prd"
  ]
  type = "api"

  request_basicauth {
    password = ""
    username = "" 
  }

}

```

**Comment by chussenot** [comment]
I close the issue
